internal/service/source.go

```diff
@@ -161,30 +161,109 @@ func (h *ServiceHandler) UpdateSource(ctx context.Context, request server.Update
 	if agent == nil {
 		newAgent := model.NewAgentForSource(uuid.New(), *source)
 		if _, err := h.store.Agent().Create(ctx, newAgent); err != nil {
-			return server.UpdateSource500JSONResponse{}, nil
+			return server.UpdateSourceInventory500JSONResponse{}, nil
 		}
 	}
 
 	user := auth.MustHaveUser(ctx)
 	if source.OrgID != user.Organization {
-		return server.UpdateSource403JSONResponse{}, nil
+		return server.UpdateSourceInventory403JSONResponse{}, nil
 	}
 
 	if source.VCenterID != "" && source.VCenterID != request.Body.Inventory.Vcenter.Id {
 		_, _ = store.Rollback(ctx)
-		return server.UpdateSource400JSONResponse{}, nil
+		return server.UpdateSourceInventory400JSONResponse{}, nil
 	}
 
-	source = mappers.UpdateSourceOnPrem(source, request.Body.Inventory)
+	source = mappers.UpdateSourceFromApi(source, request.Body.Inventory)
 
 	if _, err = h.store.Source().Update(ctx, *source); err != nil {
 		_, _ = store.Rollback(ctx)
-		return server.UpdateSource500JSONResponse{}, nil
+		return server.UpdateSourceInventory500JSONResponse{}, nil
 	}
 
 	if _, err := store.Commit(ctx); err != nil {
-		return server.UpdateSource500JSONResponse{}, nil
+		return server.UpdateSourceInventory500JSONResponse{}, nil
 	}
 
-	return server.UpdateSource200JSONResponse{}, nil
+	return server.UpdateSourceInventory200JSONResponse(mappers.SourceToApi(*source)), nil
+}
+
+func (h *ServiceHandler) UpdateSourceMetadata(ctx context.Context, request server.UpdateSourceMetadataRequestObject) (server.UpdateSourceMetadataResponseObject, error) {
+	ctx, err := h.store.NewTransactionContext(ctx)
+	if err != nil {
+		return server.UpdateSourceMetadata500JSONResponse{Message: "Failed to start transaction: " + err.Error()}, nil
+	}
+
+	source, err := h.store.Source().Get(ctx, request.Id) // Get preloads ImageInfra
+	if err != nil {
+		_, _ = store.Rollback(ctx)
+		if errors.Is(err, store.ErrRecordNotFound) {
+			return server.UpdateSourceMetadata404JSONResponse{Message: "Source not found"}, nil
+		}
+		return server.UpdateSourceMetadata500JSONResponse{Message: "Failed to get source: " + err.Error()}, nil
+	}
+
+	user := auth.MustHaveUser(ctx)
+	if source.OrgID != user.Organization {
+		_, _ = store.Rollback(ctx)
+		return server.UpdateSourceMetadata403JSONResponse{Message: "Forbidden"}, nil
+	}
+
+	updated := false
```

Comment:
I would check if the body has anything defined, and if not, return earlier instead of setting this variable all the time.
Even if the values are the same, request.Body.Name == source.Name ` I think it's worth the trouble to check it. 
This is an endpoint that does not add great loading.